node/connection_base.js

```diff
@@ -197,7 +197,8 @@ TChannelConnectionBase.prototype.resetAll = function resetAll(err) {
     outOpKeys.forEach(function eachOutOp(id) {
         var req = self.requests.out[id];
         delete self.requests.out[id];
-        // TODO: shared mutable object... use Object.create(err)?
+        err = Object.create(err);
+        err.connectionClosing = true;
```

Comment:
We have `WrappedError` for wrapping errors and adding context; I recommend using that.